unit_testing/Filtering/notchfilter_tests.cpp

```diff
@@ -0,0 +1,172 @@
+/*!
+** @file   notchfilter_tests.cpp
+** @author Matias
+** @date   23/04/2024
+*/
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <cmath>
+
+/* Fakes */
+
+/* Real supporting units */
+#include "array-math.h"
+
+/* UUT */
+#include "notchFilter.h"
+
+using namespace std;
+
+/***************************************************************************************************
+** TEST FIXTURES
+***************************************************************************************************/
+
+class NotchFilterTest: public ::testing::Test 
+{
+    protected:
+        /*******************************************************************************************
+        ** METHODS
+        *******************************************************************************************/
+        NotchFilterTest() {}
+
+        static void generateSine(float* pData, int noOfSamples, float amplitude, float freq, float fs)
```

Comment:
Probably not worth it, but you could include this in a separate file, to reduce duplication.
Filtertest_lib.c or something

Reply:
Yes I was thinking about the generateSine function being duplicated the type differs between the filters and the Goertzel however, but I guess a file containing multiple is fine. There is even another implementation in the CAProtocol test.